File: 2022/Day 1/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"time"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func part1(input_file []string) int {
	total := 0
	totals := []int{}
	for i := 0; i < len(input_file); i++ {
		if input_file[i] == "" {
			totals = append(totals, total)
			total = 0
		} else {
			intVar, err := strconv.Atoi(input_file[i])
			check(err)
			total += intVar
		}
	}
	totals = append(totals, total)

	sort.Sort(sort.Reverse(sort.IntSlice(totals)))

	// sum of totals array
	sum := 0
	for i := 0; i < 1; i++ {
		sum += totals[i]
	}

	return sum
}

func part2(input_file []string) int {
	total := 0
	totals := []int{}
	for i := 0; i < len(input_file); i++ {
		if input_file[i] == "" {
			totals = append(totals, total)
			total = 0
		} else {
			intVar, err := strconv.Atoi(input_file[i])
			check(err)
			total += intVar
		}
	}
	totals = append(totals, total)

	sort.Sort(sort.Reverse(sort.IntSlice(totals)))

	// sum of totals array
	sum := 0
	for i := 0; i < 3; i++ {
		sum += totals[i]
	}

	return sum
}

func main() {
	f, err := os.Open("./input.txt")
	check(err)

	fileScanner := bufio.NewScanner(f)
	fileScanner.Split(bufio.ScanLines)
	array_of_strings := []string{}

	for fileScanner.Scan() {
		array_of_strings = append(array_of_strings, fileScanner.Text())
	}
	start := time.Now()
	fmt.Println(part1(array_of_strings))
	elapsed := time.Since(start)

	fmt.Printf("Part 1 took %s \n", elapsed)

	start = time.Now()
	fmt.Println(part2(array_of_strings))
	elapsed = time.Since(start)

	fmt.Printf("Part 2 took %s \n", elapsed)

	f.Close()
}

File: 2022/Day 1/day1.py
"""
"""
import time
import sys

PATH="/Users/fredrikburmester/Documents/GitHub/advent-of-code"
DAY=1
INPUT_PATH=f"{PATH}/2022/Day {DAY}/input.txt"

sys.path.append(PATH)  
from utils.helpers import *


def part1(input_list):
    """ Part 1 """
    total = 0
    largest_total = 0
    for line in input_list:
        if line == '':
            if(total > largest_total):
                largest_total = total
            total = 0
        else:
            total += int(line)


    return largest_total


def part2(input_list):
    """ Part 2 """
    total_sums = []
    total = 0

    for line in input_list:
        if line == '':
            total_sums.append(total)
            total = 0
        else:
            total += int(line)

    total_sums.sort(reverse=True)
    return sum(total_sums[0:3])


def main():
    with open(INPUT_PATH, "r", encoding='UTF-8') as file:
        input_list = [str(line.strip()) for line in file]

    t0 = time.time()
    result = part1(input_list)
    t1 = time.time()
    print(f"\n{bcolors.OKGREEN}{result}{bcolors.ENDC} is the result of part 1.\n{bcolors.OKBLUE}{round(t1-t0, 4)}{bcolors.ENDC} seconds\n")

    t0 = time.time()
    result2 = part2(input_list)
    t1 = time.time()
    print(f"\n{bcolors.OKGREEN}{result2}{bcolors.ENDC} is the result of part 2.\n{bcolors.OKBLUE}{round(t1-t0, 4)}{bcolors.ENDC} seconds\n")


if __name__ == "__main__":
    main()
